src/blueprints/tv2/tv2-on-timeline-generate-service.ts

```diff
@@ -281,6 +293,47 @@ export class Tv2OnTimelineGenerateService implements BlueprintOnTimelineGenerate
     }
   }
 
+  private updateTriCasterProgramWithMediaPlayer(timelineObject: Tv2BlueprintTimelineObject, mediaPlayer: Tv2MediaPlayer): void {
+    if (timelineObject.content.deviceType !== DeviceType.TRICASTER || timelineObject.content.type !== TriCasterType.ME) {
+      return
+    }
+    const triCasterMeTimelineObject: TriCasterMixEffectTimelineObject = timelineObject as TriCasterMixEffectTimelineObject
+    if (triCasterMeTimelineObject.content.me.type !== TriCasterMixEffectContentType.PROGRAM) {
+      return
+    }
+    triCasterMeTimelineObject.content.me.programInput = this.prefixSourceInputWithTriCasterPrefix(mediaPlayer.videoMixerSource)
+  }
+
+  private prefixSourceInputWithTriCasterPrefix(sourceInput: number): TriCasterInputName {
+    return `input${sourceInput}`
+  }
+
+  private updateTriCasterLookaheadWithMediaPlayer(timelineObject: Tv2BlueprintTimelineObject, mediaPlayer: Tv2MediaPlayer): void {
+    if (timelineObject.content.deviceType !== DeviceType.TRICASTER || timelineObject.content.type !== TriCasterType.MIX_OUTPUT) {
+      return
+    }
+    const triCasterMixOutputTimelineObject: TriCasterMixOutputTimelineObject = timelineObject as TriCasterMixOutputTimelineObject
```

Comment:
If the conditions are given a signature, the type assumptions can be removed. E.g. by creating a type guard method that uses type overloading to specify the type based on the content type:

```typescript
  private isTricasterTimelineObject(timelineObject: Tv2BlueprintTimelineObject, contentType: TriCasterType.MIX_OUTPUT): timelineObject is TriCasterMixOutputTimelineObject
  private isTricasterTimelineObject(timelineObject: Tv2BlueprintTimelineObject, contentType: TriCasterType.ME): timelineObject is TriCasterMixEffectTimelineObject
  private isTricasterTimelineObject(timelineObject: Tv2BlueprintTimelineObject, contentType: TriCasterType): timelineObject is TriCasterMixOutputTimelineObject | TriCasterMixEffectTimelineObject {
    return timelineObject.content.deviceType === DeviceType.TRICASTER  && timelineObject.content.type === contentType
  }
```

Then this method can be boiled down to:
```typescript
  private updateTriCasterLookaheadWithMediaPlayer(timelineObject: Tv2BlueprintTimelineObject, mediaPlayer: Tv2MediaPlayer): void {
    if (!this.isTricasterTimelineObject(timelineObject, TriCasterType.MIX_OUTPUT)) {
      return
    }
    timelineObject.content.source = this.prefixSourceInputWithTriCasterPrefix(mediaPlayer.videoMixerSource)
  }
```

Reply:
Nice suggestion :)
I have updated both methods to use this.